components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1218,8 +1221,10 @@ void AdsServiceImpl::MaybeServeInlineContentAd(
                                    /*inline_content_ad*/ std::nullopt);
   }
 
-  bat_ads_associated_remote_->MaybeServeInlineContentAd(dimensions,
-                                                        std::move(callback));
+  bat_ads_associated_remote_->MaybeServeInlineContentAd(
+      dimensions,
+      mojo::WrapCallbackWithDefaultInvokeIfNotRun(
```

Comment:
We use `mojo::WrapCallbackWithDefaultInvokeIfNotRun` to wrap `mojo` callback (or a wrapper around `mojo` callback). This allows us call the callback and avoid DCHECK:

```
    DCHECK(!connected)
        << "BraveNewsController::GetDisplayAd was destroyed without "
        << "first either being run or its corresponding binding being closed. "
        << "It is an error to drop response callbacks which still correspond "
        << "to an open interface pipe.";
```